### PR TITLE
3.0.0 RC

### PR DESCRIPTION
**Pull request by cdaringe**
**tldr: this PR is large.  we can setup core-crew talky to go over this change if desired.**

This PR fixes https://github.com/AmpersandJS/ampersand-select-view/issues/5.  It addresses a variety of other opens as well.

Per the new README.md changelog:

## changelog

1. Improve general option edge cases, and add supporting test cases.  Primarily targets falsy option value handling.
1. Validate immediately to assist when parent FormView tests onload for field validity.  Update `skipValidation` to `skipValidationMessage`, permit immediate validation, but conditionally display messages.
1. Throw an `Error` when trying to `setValue(value)` and an option *matching the requested `value`* does not exist.  The exception to this is when the provided value is `null`, `undefined`, or `''`, and a `null` option value exists.  Because the DOM can only recognize a single empty value for any `<option>`, which is the empty string `''`, only a single empty-ish option can only be supported by the view.
1. Support `0` value options, both in Model id's and array values.
1. Add `eagerValidate`.
1. Denote a plan for 4.x release
1. bulk update README, and some cody tidying

### CODE REVIEW

**Comment by kamilogorek** [comment]
Wooh! Such a great work out here @cdaringe!
You're right that this PR is huge, but after quick scan everything looks really good.

**Comment by cdaringe** [comment]
thanks @kamilogorek.  what did you think of [this class change](https://github.com/AmpersandJS/ampersand-select-view/pull/36/files#diff-cd18a4507088002c17b5ed65ce48e6c1R45)?  I'm not 100% on that.

**Comment by wraithgar** [comment]
Typically you have a single class name for things like that, regardless of input type.  See things like bootstrap or foundation for examples.

**Comment by cdaringe** [comment]
thanks gar.  i reverted those classes back

File: README.md [hunk 1]
```diff
@@ -68,14 +125,32 @@ module.exports = FormView.extend({
...
+1. Throw an `Error` when trying to `setValue(value)` and an option *matching the requested `value`* does not exist.  The exception to this is when the provided value is `null`, `undefined`, or `''`, and a `null` option value exists.  Because the DOM can only recognize a single empty value for any <option>, which is the empty string `''`, only a single empty-ish option can only be supported by the view.
+1. Support `0` value options, both in Model id's and array values.
+1. Add `eagerValidate`.
+1. Denote a plan for 4.x release
+1. bulk update README, and some cody tidying
+
 ## credits
 
 Written by [@philip_roberts](twitter.com/philip_roberts).
+Lead Maintainer: [Christopher Dieringer (@cdaringe)](https://github.com/cdaringe)
```
Comment (wraithgar) [inline README.md]: We're putting lead right under the title, a la hapijs style.

File: package.json [hunk 2]
```diff
@@ -1,12 +1,13 @@
 {
   "name": "ampersand-select-view",
   "description": "A view module for intelligently rendering and validating selectbox input. Works well with ampersand-form-view.",
-  "version": "2.4.2",
+  "version": "3.0.0",
```
Comment (wraithgar) [inline package.json]: I try to let npm cli bump the version number, so this line can be reverted.

**Comment by wraithgar** [comment]
Travis is having problems today, tests pass fine locally.

**Comment by wraithgar** [comment]
Published as v3.0.0
